### ISSUE TITLE
🐛🐛 session does not persist using LocalAuth id 🐛🐛

### ISSUE THREAD

**Issue by andresmv94** [pedroslopez/whatsapp-web.js#2164]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have been using "whatsapp-web.js" in my application for about 6 months, and yesterday the session persistence stopped working, I am using `authStrategy: new LocalAuth({ clientId: user._id })`, and with this When I want to use the WhatsApp channel again, I can use it without having to scan the QR again, but every time that I want to use the channel and send messages, it is making me to scan the QR and this was not happening until yesterday

### Expected behavior

for first time scan the QR once, and be able to continue using whatsapp without having to scan each time I use it

### Steps to Reproduce the Bug or Issue

1. configure the server in linux to work
2. configure whatsapp web js with `authStrategy: new LocalAuth({ clientId: "some-client-id" })`
4. scan the QR and send a message
5. reuse sending messages with the same "some-client-id"

### Relevant Code

```
function authWhatsapp (user) => {
	console.log("init authentication");
	const client = await new Client({
      authStrategy: new LocalAuth({ clientId: user._id }), // user = {_id: "1234abc", name: "juan"}
      qrMaxRetries: 3,
      authTimeoutMs: 60000,
      puppeteer: {
        executablePath: process.env.CHROME_EXECUTABLE_PATH, // CHROME_EXECUTABLE_PATH=/usr/bin/google-chrome-stable
        args: ["--no-sandbox", "--disable-setuid-sandbox"],
      },
    });	
	
	client.on("qr", async (qr) => {
		console.log("scan qr");
		console.log(qr);		
		
    });
	
	 client.on("ready", async () => {
		console.log("Wapp web connected");
	 });
	
}

function sendWhatsapp (user) => {
	console.log("try send messages");
	client = await new Client({
      authStrategy: new LocalAuth({ clientId: user._id }), // user = {_id: "1234abc", name: "juan"}
      authTimeoutMs: 60000,
      puppeteer: {
        executablePath: process.env.CHROME_EXECUTABLE_PATH,  // CHROME_EXECUTABLE_PATH=/usr/bin/google-chrome-stable
        args: ["--no-sandbox", "--disable-setuid-sandbox"],
      },
    });
	
	 client.on("ready", async () => { // this is never executed 🫤
		console.log("perfect, your are authenticated, Wapp web connected");
		/*
			code to send messages...
		*/
	 });
	 
	 client.on("qr", async (qr) => {
		
		console.log("The whatsapp is asking for QR you are not authenticated");	
		console.log(qr);	
		
    });

}

for authWhatsapp function:

> init authentication
> scan qr
> 2@Zy5JBs9MeMX0GBeDlNd1....RUI9lbeATZGDNBtx52+29CIw=
> Wapp web connected

for sendWhatsapp function: 

> try send messages
> The whatsapp is asking for QR you are not authenticated
> 2@TYQbrQCmYVbCB0KG...dEy8edXIPh32AiME0=
```

### Browser Type

Google Chrome

### WhatsApp Account Type

Standard

### Does your WhatsApp account have multidevice enabled?

Yes, I am using Multi Device

### Environment

    OS: Ubuntu 20
    Phone OS:  Android, iOS
    whatsapp-web.js version [1.19.5]
    WhatsApp Web version [2.2317.10]:
    Node.js Version [v19.8.1]

### Additional context

remember: this was not happening until yesterday, I already tried setting all the permissions to the .wwebjs_auth folder, and to all its content, but it's still the same, I also notice that in windows (no server -> my personal computer) with exactly the same configuration but using `CHROME_EXECUTABLE_PATH=C:/Program Files/Google/Chrome /Application/chrome.exe`, it is working correctly, any idea why?

maybe the session files are not being saved correctly in the ".wwebjs_auth" folder, but I don't really know how to look at that part

## Update: in other computer with Windows is happenig the same issue so is not the OS.

**Comment by benjachiban** [comment]
I am experiencing the same problem.
I have been using it for a few weeks without a problem and since yesterday it has been disconnecting me often and it does not notify me why it happens. Anyone else with this error?

**Comment by andresmv94** [comment]
The whatsapp version that whatsapp-web.js mark like "working with" is = WhatsApp_Web 2.2306.7, and Now the current WhatsApp Web version  is 2.2317.10. maybe something with session was changed 

> I am experiencing the same problem. I have been using it for a few weeks without a problem and since yesterday it has been disconnecting me often and it does not notify me why it happens. Anyone else with this error?

**Comment by benjachiban** [comment]
> The whatsapp version that whatsapp-web.js mark like "working with" is = WhatsApp_Web 2.2306.7, and Now the current WhatsApp Web version is 2.2317.10. maybe something with session was changed
> 
> > I am experiencing the same problem. I have been using it for a few weeks without a problem and since yesterday it has been disconnecting me often and it does not notify me why it happens. Anyone else with this error?

Pudiste encontar solucion??

**Comment by benjachiban** [comment]
At first I found a solution, for now it's working. In the package.json inside node_modules/whatsapp-web... change the version of puppeteer to 18.1.0

**Comment by andresmv94** [comment]
> At first I found a solution, for now it's working. In the package.json inside node_modules/whatsapp-web... change the version of puppeteer to 18.1.0

Is weird because that works some times, are you using the clientId to saved the session? is in your code that clientId only numbers ex: "123456" or "1a2b3c" ?

**Comment by RC047** [comment]
same here

**Comment by samuelfaj** [comment]
Same here

**Comment by mahmouds12** [comment]
> benjachiban

I tried it, didn't work

**Comment by ch9ch9n** [comment]
Same problem

**Comment by andresmv94** [comment]
@pedroslopez , some idea about this?

**Comment by Neotastisch** [comment]
same here

**Comment by lanstat** [comment]
> At first I found a solution, for now it's working. In the package.json inside node_modules/whatsapp-web... change the version of puppeteer to 18.1.0

It worked for me, update puppeteer to 20.3.0

**Comment by XxQuickSilverZz** [comment]
Solution??

**Comment by iAndreas** [comment]
My puppeteer is on version 20.3.0 but the error persists.

**Comment by XxQuickSilverZz** [comment]
@pedroslopez Maybe your help?

**Comment by XxQuickSilverZz** [comment]
There are people who really have this in important places in the field and really no one helps... I do not understand it.

**Comment by Santosl2** [comment]
up

**Comment by Davixe7** [comment]
UP, session persistence is super unconsistent.

**Comment by longgtruong** [comment]
same problem

**Comment by daniloavilez** [comment]
I had the same problem with session in another library, it seems this problem is related to some thing that Whatsapp changed. Maybe it's a false impression I'm having testing different libraries.

**Comment by leandrosantoss** [comment]
Hi, I resolved it with permission chmod on both  .wwebjs_auth and .wwebjs_cache directories

regards

**Comment by bryanfok** [comment]
> LocalAuth

chmod 777 .wwebjs_cache 

still ask to scan qr code

**Comment by bryanfok** [comment]
> Hi, I resolved it with permission chmod on both .wwebjs_auth and .wwebjs_cache directories
> 
> regards
chmod 777 .wwebjs_cache

still ask to scan qr code

**Comment by mahmouds12** [comment]
still happening

**Comment by amoyano** [comment]
every restart localauth not working , and send new QR, any solution to this?

**Comment by myanpetra99** [comment]
> every restart localauth not working , and send new QR, any solution to this?

Have you tried the solutions mentioned above? Try updating Puppeteer to version 20.3.0 in Node_modules/whatsapp-web.js/package.json and then run 'npm update'. This worked for me.

**Comment by justoverclockl** [comment]
why this is closed if actually not work?

**Comment by alechkos** [comment]
the solution is to wait a couple minutes after the first qr code scan to let the session be "populated" properly 
also as mentioned [here])https://github.com/pedroslopez/whatsapp-web.js/issues/2164#issuecomment-1884794534) the session is being saved in `.wwebjs_auth`, so make sure that folder is created properly

**Comment by justoverclockl** [comment]
I scan the qrcode, I get the qrcode, authentications is fine…if after 2 hour I restart the server I need to rescan qrcode, so it’s not a problem related to “wait some minutes”

**Comment by justoverclockl** [comment]
i've managed this and seems to work but i get

```
ProtocolError: Waiting for selector `[data-icon='search']` failed: Runtime.callFunctionOn timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed.
1|app       |     at <instance_members_initializer> (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:86:14)
1|app       |     at new Callback (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:90:16)
1|app       |     at CallbackRegistry.create (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:22:26)
1|app       |     at Connection._rawSend (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/Connection.js:80:26)
1|app       |     at CdpCDPSession.send (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/CDPSession.js:66:33)
1|app       |     at #evaluate (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:201:50)
1|app       |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
1|app       |     at async ExecutionContext.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:168:16)
1|app       |     at async IsolatedWorld.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/IsolatedWorld.js:115:16)
1|app       |     at async CdpJSHandle.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/api/JSHandle.js:150:20)
1|app       |     at async WaitTask.rerun (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/WaitTask.js:103:28)
```

**Comment by amoyano** [comment]
> i've managed this and seems to work but i get
> 
> ```
> ProtocolError: Waiting for selector `[data-icon='search']` failed: Runtime.callFunctionOn timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed.
> 1|app       |     at <instance_members_initializer> (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:86:14)
> 1|app       |     at new Callback (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:90:16)
> 1|app       |     at CallbackRegistry.create (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:22:26)
> 1|app       |     at Connection._rawSend (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/Connection.js:80:26)
> 1|app       |     at CdpCDPSession.send (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/CDPSession.js:66:33)
> 1|app       |     at #evaluate (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:201:50)
> 1|app       |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> 1|app       |     at async ExecutionContext.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:168:16)
> 1|app       |     at async IsolatedWorld.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/IsolatedWorld.js:115:16)
> 1|app       |     at async CdpJSHandle.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/api/JSHandle.js:150:20)
> 1|app       |     at async WaitTask.rerun (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/WaitTask.js:103:28)
> ```

Check here https://github.com/pedroslopez/whatsapp-web.js/issues/2477

**Comment by justoverclockl** [comment]
@amoyano thx, done but not work, the session seems to be restored but after a while i get the above error

**Comment by justoverclockl** [comment]
i have github:pedroslopez/whatsapp-web.js#main

```javascript
const authStrategy = new LocalAuth({
  clientId: "adminSession"
});

const client = new Client({
  authStrategy,
  restartOnAuthFail: true,
  takeoverOnConflict: true,
  puppeteer: {
    headless: true,
    args: ["--no-sandbox"],
  }
});

client.initialize();

client.on("ready", () => {
  console.log("Client pronto");
});

client.on("authenticated", () => {
  console.log("Whatsapp ok");
});

client.on("disconnected", (reason) => {
  console.log("Disconnected:", reason);
});


router.get("/whatsapp/auth", async (req, res) => {
  const clientState = await client?.getState();
  console.log('ClientState: ', clientState)
  try {
    if (clientState !== "CONNECTED") {
      res.status(200).json({
        message:
          "Scan qr code to authenticate."
      });
    } 
    res.status(200).json({
        message: "Login ok."
      });
  } catch (err) {
    res.status(404).send({
      error: "Test error."
    });
  }
});
```

**Comment by justoverclockl** [comment]
its not the only broken, i keep getting

```
ProtocolError: Waiting for selector `[data-icon="search"]` failed: Runtime.callFunctionOn timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed.
0|app  |     at <instance_members_initializer> (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:86:14)
0|app  |     at new Callback (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:90:16)
0|app  |     at CallbackRegistry.create (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/CallbackRegistry.js:22:26)
0|app  |     at Connection._rawSend (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/Connection.js:80:26)
0|app  |     at CdpCDPSession.send (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/CDPSession.js:66:33)
0|app  |     at #evaluate (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:201:50)
0|app  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
0|app  |     at async ExecutionContext.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:168:16)
0|app  |     at async IsolatedWorld.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/IsolatedWorld.js:115:16)
0|app  |     at async CdpJSHandle.evaluateHandle (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/api/JSHandle.js:150:20)
0|app  |     at async WaitTask.rerun (/var/www/node_modules/whatsapp-web.js/node_modules/puppeteer-core/lib/cjs/puppeteer/common/WaitTask.js:103:28)
```

i've tried with [data-icon="search"] , i tried also const INTRO_IMG_SELECTOR = 'div[role=\'textbox\']';...but nothing...please do not close issues this fast....

**Comment by josuelmm** [comment]
Same error.. Any solution

**Comment by tuke307** [comment]
same for me. 
the library appears to have problems with multiple projects.

I have two projects: 
* App (start)
* Library (util functions)

when the client init is in the library project it cant restore the session, even if i directly specify it.

**Comment by justoverclockl** [comment]
same for me, i think something need to be updated

**Comment by gustavorps** [comment]
+1

**Comment by yairmelamed** [comment]
WhatsApp-web.js suddenly logs out on restore session with no reason.. How to fix this?

**Comment by justoverclockl** [comment]
> WhatsApp-web.js suddenly logs out on restore session with no reason.. How to fix this?

@yairmelamed we need to wait for a fix...

**Comment by AgustinParraquini** [comment]
same problem here!

**Comment by mylmz10** [comment]
+1

**Comment by justoverclockl** [comment]
any news on that?

**Comment by joaopaulopresa** [comment]
same here

**Comment by joaopaulopresa** [comment]
> At first I found a solution, for now it's working. In the package.json inside node_modules/whatsapp-web... change the version of puppeteer to 18.1.0

work for me, but some times fails

**Comment by Avivbens** [comment]
We really need some solution in here...

I'm willing to assist, just give me the details of the potential issue 🙏

**Comment by Avivbens** [comment]
OK, I've managed to resolve the issue:

**TL;DR**
Set the `webVersion` to `2.2306.7`

### NestJS provider

```
export const WhatsappClientProvider: Provider<Client> = {
    provide: WHATSAPP_CLIENT,
    inject: [ConfigService],
    useFactory: async (configService: ConfigService) => {
        const logger: Logger = new Logger('WHATSAPP_CLIENT')
        const env = configService.get<Envs>(EnvVariables.NODE_ENV, Envs.Production)
        const args = env === Envs.Production ? ['--no-sandbox', '--disable-setuid-sandbox'] : undefined

        const client = new Client({
            authStrategy: new LocalAuth(),
            webVersion: '2.2306.7', // this keeps the Whatsapp Verison with the latest version worked 👍 
            puppeteer: {
                args,
            },
        })

        client.once('qr', (qr) => {
            logger.debug('QR code received, scan please')
            return qrcode.generate(qr)
        })

        const prm = new Promise<void>((resolve) =>
            client.on('ready', async () => {
                logger.debug('Authenticated successfully!')
                resolve()
            }),
        )

        client.initialize()

        await prm
        return client
    },
}
```

### Terminal Screenshot
![CleanShot 2024-02-19 at 17 35 08](https://github.com/pedroslopez/whatsapp-web.js/assets/77183871/03db9eac-e5d2-42d9-9c3a-0fc2d97a5bcd)

**Comment by eduardotorresdev** [comment]
@Avivbens, I tried but it didn't made it. Can you share the whatsapp-web.js version? Did you change the puppeteer version? Any help is very welcome.

**Comment by Avivbens** [comment]
> @Avivbens, I tried but it didn't made it. Can you share the whatsapp-web.js version? Did you change the puppeteer version? Any help is very welcome.

Sure man 👍 


Here are my deps versions:
![CleanShot 2024-02-20 at 19 16 30](https://github.com/pedroslopez/whatsapp-web.js/assets/77183871/e353f881-a1b2-439c-ab40-c779909a680c)

**Comment by Avivbens** [comment]
@eduardotorresdev 
BTW, let the connection to establish for a minute or two...

You'll see on your WhatsApp client over the device that your chats are in sync. Keep it that way for a bit, then test the restart.

**Comment by eduardotorresdev** [comment]
@Avivbens It's worked! Thank you. I just waited the time for WhatsApp client sync with device and, that's it, worked like charm.

**Comment by Avivbens** [comment]
> @Avivbens It's worked! Thank you. I just waited the time for WhatsApp client sync with device and, that's it, worked like charm.

Happy to hear 🥳 

For future documentation - list here the versions you're on please 🙏

**Comment by eduardotorresdev** [comment]
I'm using whatsapp-web.js at "^1.23.0" version. I didn't changed puppeteer version or of any related package to wwebjs lib. I just added the webVersion that you told previously.

**Comment by eduardotorresdev** [comment]
@Avivbens Can i colaborate with something? I believe that is very important to documentate all this question of auth and his persistance, so maybe we can do this together.

**Comment by yairmelamed** [comment]
> OK, I've managed to resolve the issue:
> 
> **TL;DR** Set the `webVersion` to `2.2306.7`
> 
> ### NestJS provider
> ```
> export const WhatsappClientProvider: Provider<Client> = {
>     provide: WHATSAPP_CLIENT,
>     inject: [ConfigService],
>     useFactory: async (configService: ConfigService) => {
>         const logger: Logger = new Logger('WHATSAPP_CLIENT')
>         const env = configService.get<Envs>(EnvVariables.NODE_ENV, Envs.Production)
>         const args = env === Envs.Production ? ['--no-sandbox', '--disable-setuid-sandbox'] : undefined
> 
>         const client = new Client({
>             authStrategy: new LocalAuth(),
>             webVersion: '2.2306.7', // this keeps the Whatsapp Verison with the latest version worked 👍 
>             puppeteer: {
>                 args,
>             },
>         })
> 
>         client.once('qr', (qr) => {
>             logger.debug('QR code received, scan please')
>             return qrcode.generate(qr)
>         })
> 
>         client.on('authenticated', () => {
>             logger.debug('Authenticated successfully!')
>         })
> 
>         client.on('disconnected', (reason) => {
>             logger.error(`Disconnected, reason: ${reason}`)
>         })
> 
>         const prm = new Promise<void>((resolve) =>
>             client.on('ready', async () => {
>                 logger.debug('WhatsApp Client is ready!')
>                 resolve()
>             }),
>         )
> 
>         client.initialize()
> 
> 
>         await prm
>         return client
>     },
> }
> ```
> 
> ### Terminal Screenshot
> ![CleanShot 2024-02-19 at 17 35 08](https://private-user-images.githubusercontent.com/77183871/306006976-03db9eac-e5d2-42d9-9c3a-0fc2d97a5bcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1OTA5ODYsIm5iZiI6MTcwODU5MDY4NiwicGF0aCI6Ii83NzE4Mzg3MS8zMDYwMDY5NzYtMDNkYjllYWMtZTVkMi00MmQ5LTljM2EtMGZjMmQ5N2E1YmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIyVDA4MzEyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxMGRjZmYzMTZjNmM4NDdmMDgzOGVkZTdmMGQ3ZjJjYWEzY2M4NTczMDYyZWMzNmY5NmNhNGI2MjU0MWNmZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0fgbM04oGH12OO7hhosmZYDx1AI-iCcs9B8AfMQydys)


It doesn't help, it works for one week and then suddenly it disconnects and asks for qr again with no reason.. how can we fix this?? why it's happening?

**Comment by Avivbens** [comment]
> > OK, I've managed to resolve the issue:
> > **TL;DR** Set the `webVersion` to `2.2306.7`
> > ### NestJS provider
> > ```
> > export const WhatsappClientProvider: Provider<Client> = {
> >     provide: WHATSAPP_CLIENT,
> >     inject: [ConfigService],
> >     useFactory: async (configService: ConfigService) => {
> >         const logger: Logger = new Logger('WHATSAPP_CLIENT')
> >         const env = configService.get<Envs>(EnvVariables.NODE_ENV, Envs.Production)
> >         const args = env === Envs.Production ? ['--no-sandbox', '--disable-setuid-sandbox'] : undefined
> > 
> >         const client = new Client({
> >             authStrategy: new LocalAuth(),
> >             webVersion: '2.2306.7', // this keeps the Whatsapp Verison with the latest version worked 👍 
> >             puppeteer: {
> >                 args,
> >             },
> >         })
> > 
> >         client.once('qr', (qr) => {
> >             logger.debug('QR code received, scan please')
> >             return qrcode.generate(qr)
> >         })
> > 
> >         client.on('authenticated', () => {
> >             logger.debug('Authenticated successfully!')
> >         })
> > 
> >         client.on('disconnected', (reason) => {
> >             logger.error(`Disconnected, reason: ${reason}`)
> >         })
> > 
> >         const prm = new Promise<void>((resolve) =>
> >             client.on('ready', async () => {
> >                 logger.debug('WhatsApp Client is ready!')
> >                 resolve()
> >             }),
> >         )
> > 
> >         client.initialize()
> > 
> > 
> >         await prm
> >         return client
> >     },
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ### Terminal Screenshot
> > ![CleanShot 2024-02-19 at 17 35 08](https://private-user-images.githubusercontent.com/77183871/306006976-03db9eac-e5d2-42d9-9c3a-0fc2d97a5bcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1OTA5ODYsIm5iZiI6MTcwODU5MDY4NiwicGF0aCI6Ii83NzE4Mzg3MS8zMDYwMDY5NzYtMDNkYjllYWMtZTVkMi00MmQ5LTljM2EtMGZjMmQ5N2E1YmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIyVDA4MzEyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxMGRjZmYzMTZjNmM4NDdmMDgzOGVkZTdmMGQ3ZjJjYWEzY2M4NTczMDYyZWMzNmY5NmNhNGI2MjU0MWNmZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0fgbM04oGH12OO7hhosmZYDx1AI-iCcs9B8AfMQydys)
> 
> It doesn't help, it works for one week and then suddenly it disconnects and asks for qr again on restore with no reason.. how can we fix this?? why it's happening?

Well, in general, the library has an Apache license, meaning it has no responsibility for the stability of the product. 
I can just assume that Whatsapp servers can revoke client authentication based on its behavior. 


That being said, you can try to use the [RemoteAuth](https://wwebjs.dev/guide/authentication.html#remoteauth-strategy) strategy, it might be a bit more stable.

**Comment by yairmelamed** [comment]
Hello every one.. is there a solution here? It is so frustrating.. It still happens..

**Comment by josuelmm** [comment]
Same problem

**Comment by MikeSoft** [comment]
up

**Comment by Hasnen-mr** [comment]
resolved with remote auth, 
let me know if there is still an issue. 
WhatsApp me +91 8320354644

**Comment by ingGhost** [comment]
> resolved with remote auth, let me know if there is still an issue. WhatsApp me +91 8320354644

how u can fix the issue?

**Comment by AKwoKWH** [comment]
having the same issue recently

**Comment by luandro** [comment]
Also having the same issue, any updates?

**Comment by MikeSoft** [comment]
> any update?

I updated to the last version   `"whatsapp-web.js": "1.26.1-alpha.2"`  and apparently it's solved.
`const wa_client = new Client({authStrategy: new LocalAuth({ dataPath: "sessions" }),});`

**Comment by Gajasurve** [comment]
I deleted the `.wwebjs_auth` and `.wwebjs_cache` files, then logged in again. After scanning the QR code, I didn't stop the process immediately and allowed it to sync. Once the sync was complete, it no longer asked for the QR code. This worked for me, and it might work for you as well.

**Comment by proggerkz** [comment]
Is this problem solved within new update or its still open?

**Comment by devhasibulislam** [comment]
## Problem Statement
After authenticating with a QR code using LocalAuth, the client connects successfully and message sending works as expected. However, when I stop and restart the server, the session does not persist. A new QR code is generated again, and I must rescan it. This indicates that the previous session is not being restored, even though the session folder and files remain present on disk.

## Expected Behavior
As a user, I expected the WhatsApp client to automatically reconnect using the previously authenticated session without needing to scan the QR code again after a server restart, but I am seeing a new QR being generated instead.

### Source Code
```tsx
import { Client, LocalAuth } from "whatsapp-web.js";

const client = new Client({
  authStrategy: new LocalAuth({
    clientId: "some_unique_id", // unique for each user
    dataPath: "./src/modules/whatsapp/session", // persists session here
  }),
  puppeteer: {
    headless: true,
    args: ["--no-sandbox", "--disable-setuid-sandbox"],
  },
});

client.on("qr", (qr) => {
  console.log("QR Received", qr); // <-- triggers again after restart
});

client.on("ready", () => {
  console.log("Client is ready");
});

client.initialize();
```

Client session not persisting after server restart despite using `LocalAuth` | `RemoteAuth`. Is there anyone who able to solve this???

**Comment by isabellacalfa** [comment]
someone solve it? i updated to v1.31 and still not working correctly. connection stands ok for just 24h

**Comment by nihatsen** [comment]
My client disconnects in about 24hours too (with restarts of course. Session is saved remotely and it works perfectly). and starts requesting QR. Any fix?

**Comment by MikeSoft** [comment]
@nihatsen upgrade the version that your're currently using and use a local session, it helps.

**Comment by devhasibulislam** [comment]
> [@nihatsen](https://github.com/nihatsen) upgrade the version that your're currently using and use a local session, it helps.

Can you say actually which version worth only the session persist thing? I mean do think latest can do it?

**Comment by isabellacalfa** [comment]
> [@nihatsen](https://github.com/nihatsen) upgrade the version that your're currently using and use a local session, it helps.

I’m using v1.31 and keeps not working. What do you mean with local session?

**Comment by devhasibulislam** [comment]
> > [@nihatsen](https://github.com/nihatsen) upgrade the version that your're currently using and use a local session, it helps.
> 
> I’m using v1.31 and keeps not working. What do you mean with local session?

There's two types of session local (saved in device) & remote (saved in database). But it won't works with any :(

**Comment by RC047** [comment]
is there any update on this issue?

**Comment by devhasibulislam** [comment]
> is there any update on this issue?

From my side, I must say it's an unsolvable issue :( I've no comments from now.

**Comment by JuanRa012012** [comment]
Me too, I have about 5 instances up, but it's difficult to be scanning the QR almost every day

**Comment by pablocunha** [comment]
I have the same problem.

**Comment by luccasmarra** [comment]
The problem was fixed? 
What is the solution?

**Comment by brunoinds** [comment]
Same problem.

**Comment by brunoinds** [comment]
> The problem was fixed? What is the solution?

Try it:

1/ make sure .wwebjs_auth is created after login, otherwise likely your code problem
2/ puppeteer is update
3/ make sure .wwebjs_auth is chmod 777

**Comment by HimanshuInRaoInfo** [comment]
I’m trying to create a client connection with LocalAuth in my setup.
Here’s the function I’m using:

```
const createClientConnection = () => {
  client[clientId] = new Client({
    puppeteer: {
      pipe: true,
      executablePath: chromePath,
      headless: false,
      args: [
        "--no-sandbox",
        "--disable-setuid-sandbox",
        "--disable-gpu",
        "--disable-dev-shm-usage"
      ]
    },
    session: clientId, // <--- kept here for testing
    authStrategy: new LocalAuth({
      clientId: clientId,
      dataPath: databasePath,
    }),
  });

  client[clientId].initialize();
}
```

The issue is:
- When I run this, the WhatsApp Web window opens and connects fine.
- But none of the client event hooks are triggered (loading_screen, ready, authenticated, auth_failure, etc.).
- It looks like the session initializing but the whatsapp web is loaded fully.

Is this expected because I’m mixing session with LocalAuth, or am I missing something in how to correctly attach the event listeners?

version : "whatsapp-web.js": "^1.34.0"
OS : windows 10

**Comment by Walkercito** [comment]
I'm guessing there is no updates in the error so far :( 

Is there any idea of why is this happening or when exactly did it broke to look into the diff?

**Comment by dmanzanomartin** [comment]
I’m not sure whether the issue comes from the library or from WhatsApp Web. If I restart my script while my phone is on, the session persists; however, if I restart it with the phone turned off, it doesn’t work.

**Comment by Walkercito** [comment]
was it fixed?

**Comment by tonywang806** [comment]
The issue I encountered might be the same. 
The symptom is that after scanning the QR code, the session remains active for a period of time and then automatically triggers a disconnect event.

I checked WhatsApp’s Terms of Service and found that the mobile session can only be maintained for 14 days. 
Therefore, our current solution is to restart the service once a week. 
We use pm2 to manage the process, and we reload the locally saved session by running `pm2 stop` followed by `pm2 start`.

The project has been running in this way for about one year, and overall it has been working normally.

According to the[ WhatsApp FAQ](https://faq.whatsapp.com/378279804439436/?helpref=faq_content&cms_platform=android):

>Your phone doesn’t need to stay online to use WhatsApp on linked devices, but your linked devices will be logged out if you don’t use your phone for over 14 days.
